builder/simplify.cc

```diff
@@ -589,30 +653,58 @@ class simplifier : public node_mutator {
     if (!result.same_as(op)) {
       mutate_and_set_result(result);
     } else {
-      set_result(result, bounds_of(op, std::move(a_bounds), std::move(b_bounds)));
+      set_result(result, {bounds_of(op, std::move(a_info.bounds), std::move(b_info.bounds)), modulus_remainder::unify(a_info.alignment, b_info.alignment)});
     }
   }
 
   void visit(const class min* op) override { visit_min_max(op); }
   void visit(const class max* op) override { visit_min_max(op); }
 
+  modulus_remainder modulus_of(const add* op, const modulus_remainder& a, const modulus_remainder& b) { return a + b; }
+  modulus_remainder modulus_of(const sub* op, const modulus_remainder& a, const modulus_remainder& b) { return a - b; }
+  modulus_remainder modulus_of(const mul* op, const modulus_remainder& a, const modulus_remainder& b) { return a * b; }
+  modulus_remainder modulus_of(const div* op, const modulus_remainder& a, const modulus_remainder& b) { return a / b; }
+  modulus_remainder modulus_of(const mod* op, const modulus_remainder& a, const modulus_remainder& b) { return a % b; }
+
   template <typename T>
   void visit_binary(const T* op) {
-    interval_expr a_bounds;
-    expr a = mutate(op->a, &a_bounds);
-    interval_expr b_bounds;
-    expr b = mutate(op->b, &b_bounds);
+    expr_info a_info;
+    expr a = mutate(op->a, &a_info);
+    expr_info b_info;
+    expr b = mutate(op->b, &b_info);
 
     if (!a.defined() || !b.defined()) {
-      set_result(expr(), interval_expr());
+      set_result(expr(), expr_info());
       return;
     }
 
+    int a_mod = a_info.alignment.modulus;
```

Comment:
`index_t`, or in this case since it's simply giving a shorter name to the same thing, maybe `auto` is OK here.